File: app/src/main/java/com/junli/android_database/java/room/api/NormalUserDao.java
package com.junli.android_database.java.room.api;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.junli.android_database.java.room.domain.NormalUser;

import java.util.List;

@Dao
public interface NormalUserDao {
    @Insert
    void insertNormalUser(NormalUser normalUser);

    @Delete
    int deleteNormalUser(NormalUser normalUser);

    @Update
    int updateNormalUser(NormalUser normalUser);

    @Query("Update normal_user_java set nick_name=:nickName where user_id = :userID")
    void updateNormalUserNickNameByID(String nickName, Long userID);

    @Query("delete from normal_user_java where user_id = :userID")
    int deleteNormalUserByID(Long userID);

    @Query("SELECT * FROM normal_user_java where user_id= :userID")
    List<NormalUser> getNormalUserByID(Long userID);

    @Query("SELECT * FROM normal_user_java")
    List<NormalUser> getAllNormalUser();
}

File: app/src/main/java/com/junli/android_database/kotlin/room/dao/NormalUserDao.kt
package com.junli.android_database.kotlin.room.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.junli.android_database.kotlin.room.domain.NormalUser

@Dao
interface NormalUserDao {
    @Insert
    fun insertNormalUser(normalUser: NormalUser)

    @Query("SELECT * FROM normal_user WHERE user_id = :userID")
    fun selectNormaUserByID(userID: Long): NormalUser

    @Query("SELECT * FROM normal_user")
    fun selectAllNormalUser(): List<NormalUser>
}